File: 0066-plus-one/main.go
package main

import (
	"fmt"
)

func plusOne(digits []int) []int {
  var result []int
	carry := 1
	for i := (len(digits) - 1); i >= 0; i-- {
		dd := digits[i]
		dd = dd + carry
		carry = dd / 10 // integer division quotient
		dd = dd % 10
		result = append([]int{dd}, result...) // unshift
	}

	if carry > 0 {
		result = append([]int{carry}, result...) // unshift
	}

	return result
}

func main() {
	e1 := []int{1,2,3}
	fmt.Println(plusOne(e1))

	e2 := []int{4,3,2,1}
	fmt.Println(plusOne(e2))

	e3 := []int{9}
	fmt.Println(plusOne(e3))

	e4 := []int{9,9,9}
	fmt.Println(plusOne(e4))
}

File: 0066-plus-one/main.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  let result = [];
  let carry = 1;
  for (i = digits.length - 1; i >=0; i--) {
    let dd = digits[i];
    dd = dd + carry;
    carry = ~~(dd/10)  // integer division quotient
    dd = dd % 10;
    result.unshift(dd)
  }
  if (carry > 0) { result.unshift(carry) }
  return result;
};

// const e1 = [1,2,3];
// console.log(plusOne(e1));

// const e2 = [4,3,2,1];
// console.log(plusOne(e2));

// const e3 = [9];
// console.log(plusOne(e3));

// const e4 = [9,9,9];
// console.log(plusOne(e4));